packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -219,6 +219,42 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     }
   }
 
+  // Check if we can, and probably should, upgrade the (ios) simulator
+  if (Simulator.isPlatformSupported()) {
```

Comment:
I don't think there is a (public) method available to double-check if an upgrade is necessary right?

Reply:
not that i am aware of, cc @fson

Reply:
It's ok, might be a nice to have but not necessary now It think.